contributors/devel/sig-architecture/feature-gates.md

```diff
@@ -1,87 +1,256 @@
+Beta features are usually enabled by default. This allows them to be available
+in the default installation of Kubernetes. Users may switch them off
+explicitly.
 
-Sometimes (rarely) the `Default` is set to `false` in Beta. This tells folks that while this feature is Beta, they will still need to do some work to switch it on and use it, and potentially take some other explicit action outside of Kubernetes. For example, the CSIMigration feature gates looked like this:
+Sometimes (rarely) a beta feature will be disabled by default. This tells users
+that while this feature is Beta, they will still need to do some work to switch
+it on and use it, which may include taking some other explicit action outside of
+Kubernetes. For example, the CSIMigration feature gates looked like this:
 
 ```
-	CSIMigration:                                   {Default: true, PreRelease: featuregate.Beta},
+	CSIMigration:                                   {Default: true,  PreRelease: featuregate.Beta},
 	CSIMigrationGCE:                                {Default: false, PreRelease: featuregate.Beta}, // Off by default (requires GCE PD CSI Driver)
 	CSIMigrationAWS:                                {Default: false, PreRelease: featuregate.Beta}, // Off by default (requires AWS EBS CSI driver)
 	CSIMigrationAzureDisk:                          {Default: false, PreRelease: featuregate.Beta}, // Off by default (requires Azure Disk CSI driver)
 	CSIMigrationAzureFile:                          {Default: false, PreRelease: featuregate.Beta}, // Off by default (requires Azure File CSI driver)
 	CSIMigrationvSphere:                            {Default: false, PreRelease: featuregate.Beta}, // Off by default (requires vSphere CSI driver)
 ```
 
-## GA Features
+### GA features
 
 * `PreRelease` is set to `featuregate.GA`
 * `Default` is always set to `true`
-* `LockToDefault` is set to `true`
+* `LockToDefault` is usually set to `true` (see below)
 
 GA features are always on by default, and usually cannot be disabled.
 
-Sometimes (rarely) we do not set `LockToDefault` (thus defaulting to `false`, meaning "not locked"). This enables folks to switch off the GA feature. We do this to indicate that while the feature (for example `coredns`) is GA, they need to move off `kubedns` to `coredns` in their infrastructure. If they want to continue using `kubedns` for a short time, they can choose to switch off the GA flag. When we eventually remove the support for `kubedns` entirely we would set `LockToDefault` to `true` with some grace period for the transition.
+Sometimes (rarely) a GA feature is allowed to be disabled (`LockToDefault:
+false`). This indicates that while this feature is GA, they need to take some
+other action outside of Kubernetes. For example, to use `CoreDNS`, users had to
+move off `kubedns` in their infrastructure. If they wanted to continue using
+`kubedns`, they could choose to switch off the GA flag. When we eventually
```

Comment:
was kubedns a feature gate?
it is an external component

Reply:
I just kept the example that was there - I agree it's not a great one.  I am tempted to just remove the example.